docs/template_merging.md

```diff
@@ -0,0 +1,147 @@
+
+The string consists of three groups separated with a pipe.
+1. [01]?[0-9]{1,2} - Matches any number between 0 and 199. [01]?: 0 or 1 may appear at most once at front of the number. [0-9]{1,2}: digits 0 to 9 may appear exactly once or twice on the 2nd or 3rd position in the number.
+2. 2[0-4][0-9] - Matches numbers between 200 and 249, where the first digit is always 2, the second is between 0 and 4, and the third digit is any between 0 and 9,
+3. 25[0-5]: (the 3rd group) matches numbers between 250 and 255, where 25 is always at front and the third digit is between 0 and 5.
+
+> **See Also:** JSON schema [Built-in Formats](https://json-schema.org/understanding-json-schema/reference/string.html?highlight=maxlength#built-in-formats) and [Regular Expressions](https://json-schema.org/understanding-json-schema/reference/string.html#id6) for more information.
+
+**Boolean:** The boolean type { type: boolean } matches two values; true or false and must be used in all lower case characters.
+
+
+# Combining Schema
+JSON uses the keywords allOf, anyOf and oneOf for combining schema together.
+BIG-IP FAST also uses they keywords of oneOf/allOf/anyOf for template merging, however this section is focused on JSON schema.
+
+**anyOf:** One or more of the contained schema is validated against the instance value.
+It is less restrictive than allOf as more than one of the same type may be specified.
+```javascript
+{
+    "anyOf": [
+        { "type": "string" },
+        { "type": "number" }
+    ]
+}
+```
+
+**oneOf:** Validates against exactly one subschema even though multiple instances listed.
+For example, if multipleOf is set to 5 and 3, validation will pass on 10 and 9, but will fail on 2 as neither 5 nor 3 are multiples of 2.
+It will also fail on 15 as it is a multipleOf both 5 and 3 not oneOf.
+```javascript
+{
+    "oneOf": [
+        { "type": "number", "multipleOf": 5 },
+        { "type": "number", "multipleOf": 3 }
+    ]
+}
+```
+
+**allOf:** All of the contained schemas must validate against the instance value.
+```javascript
+{
+    "allOf": [
+        { "type": "string" },
+        { "maxLength": 5 }
+    ]
+}
+```
+
+> **Note:** When using allOf, be cautious of specifying multiple types such as `{ type: string }` and `{ type: number }` as a type cannot be a string and a number at the same time.
+
+When authoring templates using yaml, allOf takes on a special meaning by referencing another template in the set, known as Template Merging.
+* allOf will merge the schema of the merge template with external template(s) just as JSON schema will when generating schema for the merged templates
+* When a merge template is rendered, the JSON output of the templates will be merged together
+* Merge can be used to add additional configuration to a template
+```yaml
+parameters:
+    ...
+definitions:
+    ...
+template: |
+    ...
+allOf:
+    - $ref: "tcp.yaml#"
+```
```

Comment:
This seems to be the only thing that actually discusses template merging in this page. We should start with this and expand on it.

Reply:
With the updates to the first confusing example on the template_merging.md page, we now discuss a more simple example of top-level definition merging first, and then this example -- which should probably still be expanded upon, but I have not done that. 

Should we leave this at the end and expand on it for more specific schema merging options?